### PR TITLE
Provision to create PaymentProfile using StripeToken

### PR DESCRIPTION
**Pull request by riteshshrv**
[#130268657]

### CODE REVIEW

File: party.py [hunk 1]
```diff
@@ -36,10 +40,51 @@ class PaymentProfile:
         'Stripe Customer ID', readonly=True
     )
 
+    @classmethod
+    def __setup__(cls):
+        super(PaymentProfile, cls).__setup__()
+        cls.__rpc__.update({
+            'create_stripe_profile': RPC(instantiate=0, readonly=True),
```
Comment (prakashpp) [inline party.py]: create_profile_using_stripe_token

File: party.py [hunk 2]
```diff
@@ -36,10 +40,51 @@ class PaymentProfile:
...
+    def create_stripe_profile(cls, user_id, gateway_id, token):
+        """
+        Creat a Stripe Customer Object
+        """
+        Party = Pool().get('party.party')
+        PaymentGateway = Pool().get('payment_gateway.gateway')
+
+        party = Party(user_id)
+        gateway = PaymentGateway(gateway_id)
+        stripe.api_key = gateway.stripe_api_key
```
Comment (prakashpp) [inline party.py]: assert gateway is stripe

File: party.py [hunk 3]
```diff
@@ -36,10 +40,51 @@ class PaymentProfile:
...
+            )
+            card = customer.sources.data[0]
+        except (
+            stripe.error.CardError, stripe.error.InvalidRequestError,
+            stripe.error.AuthenticationError, stripe.error.APIConnectionError,
+            stripe.error.StripeError
+        ), exc:
+            raise UserError(exc.json_body['error']['message'])
+        else:
+            return customer.id, card.id
```
Comment (prakashpp) [inline party.py]: return card.id

File: party.py [hunk 4]
```diff
@@ -36,10 +40,51 @@ class PaymentProfile:
...
+
 
 class Party:
     __name__ = 'party.party'
 
+    @classmethod
+    def __setup__(cls):
+        super(Party, cls).__setup__()
+        cls.__rpc__.update({
+            'create_stripe_profile': RPC(instantiate=0, readonly=True),
```
Comment (prakashpp) [inline party.py]: not needed here

File: party.py [hunk 5]
```diff
@@ -36,6 +40,42 @@ class PaymentProfile:
...
+        cls.__rpc__.update({
+            'create_profile_using_stripe_token': RPC(
+                instantiate=0, readonly=True
+            ),
+        })
+
+    @classmethod
+    def create_profile_using_stripe_token(cls, user_id, gateway_id, token):
+        """
+        Creat a Stripe Customer Object
```
Comment (prakashpp) [inline party.py]: typo

Comment (riteshshrv) [inline party.py]: Fixed

**Pull request merged**